lnd_test.go

```diff
@@ -16,109 +13,181 @@ import (
 	"github.com/roasbeef/btcd/wire"
 	"github.com/roasbeef/btcrpcclient"
 	"github.com/roasbeef/btcutil"
+	"google.golang.org/grpc"
+	"testing"
+	"github.com/go-errors/errors"
 )
 
-type lndTestCase func(net *networkHarness, t *testing.T)
+// CT is needed for:
+// - have uniform way of handling panic and fatal error from test cases.
+// - have ability to properly wrap errors in order to see stack trace.
+// - have nice and elegant way to handle lnd process errors in one
+// select structure with test cases.
+type CT struct {
```

Comment:
Very nice! I really like the addition of this wrapper struct. Currently, when I run into daemon related errors locally, I make some local changes to `networkHarness` to allow me to peer into the logs and debug the issue manually. With the addition of this wrapper struct, errors propagate cleanly, providing a better user experience for developers writing new integration tests.